src/ui/public/modules.js

```diff
@@ -1,9 +1,43 @@
+/**
+ * This module is used by Kibana to create and reuse angular modules. Angular modules
+ * can only be created once and need to have their dependencies at creation. This is
+ * hard/impossible to do in require.js since all of the dependencies for a module are
+ * loaded before it is.
+ *
```

Comment:
Got it. If that's what the doc was trying to convey, it wasn't clear to me. I'm having a hard time thinking of a better way to say it though. Perhaps this example should go in the doc as well?